### ISSUE TITLE
bug ?:  cant set property  '... ' of undefined 

### ISSUE THREAD

**Issue by Eric-wh-007** [mac-s-g/react-json-view#208]
### desc:  
i set src from props  
```
const { rule: { data }, form: { getFieldDecorator } } = this.props;
<ReactJson
              src={data}
              name={false}
              displayDataTypes={false}
              onEdit={this.handleJsonEdit}
              onAdd={this.handleJsonAdd}
              onDelete={this.handleDelete}
            />
```
i got this json  
```
{
"name":"wh"
"age":11
"rules":{
    "rule":"aaaa"
  }
}
```

when i try to edit the field 'rule' throw err
```
Uncaught TypeError: Cannot set property 'rule' of undefined
    at t.i.updateSrc (main.js:710)
    at t.i.handleAction (main.js:670)
    at e._invokeCallback (main.js:5427)
    at e.dispatch (main.js:5420)
    at t.submitEdit (main.js:6183)
```

 no bug when i try to edit 'name' , 'age',  but my json change to this
```
{
    "age":"11dfff"
}
```
the other fields disappear!

 if i set src like that 
```
src={{rules: {
                name:'wh',
                age:11,
                rules:{
                  rule:'aaaa',
                },
              }}}
```
no problem！

so  why this bug appear?

**Comment by danielsantossonos** [comment]
I am having the exact same error.

**Comment by vdelord** [comment]
Hi, I recognize this behavior. This is a clone from #206. I published a PR to fix this problem.
Can you try if it fix your problem ?

**Comment by mac-s-g** [comment]
@Eric-wh-007 it would be helpful if you could confirm that #206 fixes the issue.  

@vdelord thanks for the tip,  I'll take a look at your pr now.

**Comment by mac-s-g** [comment]
@Eric-wh-007 I believe the issue is resolved with 1.19.1.
